composer/trainer/trainer.py

```diff
@@ -1251,6 +1421,7 @@ def __init__(
         if parallelism_config is not None:
             # Patch PyTorch to fix distributed bugs
             patch_pytorch()
+            patch_unshard_for_automicrobatching(self.auto_microbatch_size_found)
```

Comment:
this should be just part of patch_pytorch to simplify interface

Reply:
we need to pass in a boolean variable telling it how to patch this one specific method though - i feel like it would be less readable if we passed `self.auto_microbatch_size_found` directly into `patch_pytorch`